src/backend/distributed/sql/citus--10.2-4--11.0-1.sql

```diff
@@ -30,3 +30,17 @@ BEGIN
 	END IF;
 END;
 $$;
+
+-- Here we keep track of partitioned tables that exists before Citus 11
+-- where we need to call fix_all_partition_shard_index_names() before
+-- metadata is synced. Note that after citus-11, we automatically
+-- adjust the indexes so we only need to fix existing indexes
+DO LANGUAGE plpgsql
+$$
+DECLARE
+  partitioned_table_count int :=0;
+BEGIN
+      SELECT count(*) INTO partitioned_table_count FROM pg_dist_partition p JOIN pg_class c ON p.logicalrelid = c.oid WHERE c.relkind = 'p';
```

Comment:
In a private chat, Marco noted maybe we could use a boolean as the count is not useful anyway. I think that makes sense, keeping the count could be confusing.

I'm noting it here for reference and I'll update the PR accordingly

Reply:
He also noted this here https://github.com/citusdata/citus/pull/5556#discussion_r773912130